### PR TITLE
Add example suite.yaml files for gator test

### PR DESCRIPTION
**Pull request by willbeason**
These can be run individually, or all at once with:

    gator test ./...

Signed-off-by: Will Beason <willbeason@google.com>

### CODE REVIEW

**Review by maxsmythe — APPROVED** [review]
LGTM

**Review by julianKatz — APPROVED** [review]

File: library/general/block-endpoint-edit-default-role/suite.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+kind: Suite
+apiVersion: test.gatekeeper.sh/v1alpha1
+tests:
+- name: allowed-repos
```
Comment (ritazh) [inline library/general/block-endpoint-edit-default-role/suite.yaml:4]: nit: update the test name?

File: library/general/block-nodeport-services/suite.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+kind: Suite
+apiVersion: test.gatekeeper.sh/v1alpha1
+tests:
+- name: allowed-repos
```
Comment (ritazh) [inline library/general/block-nodeport-services/suite.yaml:4]: ditto

File: library/general/containerlimits/suite.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+kind: Suite
+apiVersion: test.gatekeeper.sh/v1alpha1
+tests:
+- name: allowed-repos
```
Comment (ritazh) [inline library/general/containerlimits/suite.yaml:4]: ditto

File: library/general/allowedrepos/suite.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+kind: Suite
+apiVersion: test.gatekeeper.sh/v1alpha1
+tests:
+- name: allowed-repos
+  template: template.yaml
+  constraint: samples/repo-must-be-openpolicyagent/constraint.yaml
+  cases:
+  - name: example-allowed
+    object: samples/repo-must-be-openpolicyagent/example_allowed.yaml
```
Comment (ritazh) [inline library/general/allowedrepos/suite.yaml:9]: Question: if there's no assertion, does it default to `violations: no`? if so, when does a user want to explicitly use assertions with `violation: no`?

Comment (maxsmythe) [reply library/general/allowedrepos/suite.yaml:9]: I think this thread is most relevant:

https://github.com/open-policy-agent/gatekeeper/pull/1519#discussion_r697701073

Agreed that zero assertions => implied assertion that the object will pass validation.

IMO it should be that zero assertions => zero tests against the object => the test can never fail

Comment (maxsmythe) [reply library/general/allowedrepos/suite.yaml:9]: If we leave the behavior as-is, we risk winding up with weird edge cases if the tests become more complex (e.g. imagine we are not testing validation)

File: library/general/block-nodeport-services/suite.yaml (hunk 2 above)
Comment (willbeason) [reply library/general/block-nodeport-services/suite.yaml]: Done.

File: library/general/block-endpoint-edit-default-role/suite.yaml (hunk 1 above)
Comment (willbeason) [reply library/general/block-endpoint-edit-default-role/suite.yaml]: Done.

File: library/general/allowedrepos/suite.yaml (hunk 4 above)
Comment (ritazh) [reply library/general/allowedrepos/suite.yaml:9]: wdyt @willbeason ?

Comment (willbeason) [reply library/general/allowedrepos/suite.yaml:9]: I made the change you suggested.

Comment (ritazh) [reply library/general/allowedrepos/suite.yaml:9]: thanks! but does that mean gator doesnt require assertions?

**Review by ritazh — APPROVED** [review]

File: library/general/allowedrepos/suite.yaml (hunk 4 above)
Comment (maxsmythe) [reply library/general/allowedrepos/suite.yaml:9]: Will told me today he wanted to change it so that a test with no assertions is always invalid and will fail.
